### DISCUSSION TITLE
How to convert checkpoint to Pretrained.Interface?

### DISCUSSION THREAD

**Discussion by 21jun** [speechbrain/speechbrain#1976] [Q&A]
When we train a model with example recipes, the final result looks like this:
![Image](https://github.com/speechbrain/speechbrain/assets/29483429/b1bb6fc5-6b9e-4a4d-ba37-6e15a1797365)

However, we may want to use such a checkpoint to run simple inference, similar to the interface classes available in [SpeechBrain](https://speechbrain.readthedocs.io/en/latest/API/speechbrain.pretrained.interfaces.html):
![Image](https://github.com/speechbrain/speechbrain/assets/29483429/7634c1a1-1de4-4819-a21a-824dd01bdf02)

In summary, I want to convert the checkpoint from [this Google Drive folder](https://drive.google.com/drive/folders/19G2Zm8896QSVDqVfs7PS_W86-K0-5xeC) to the format found at [this Hugging Face repository](https://huggingface.co/speechbrain/asr-wav2vec2-commonvoice-de).

Is there any method or script available for this conversion?

Thank you in advance!
